Issue with double-quoted options

I can diagnose this further later, but you might know immediately where this is. I think parsing of options with quotes is trimming the second quote. See https://github.com/uber/prototool/issues/374.

Here's a diff that reproduces this:

```
diff --git a/option_test.go b/option_test.go
index 3188461..979f9f2 100644
--- a/option_test.go
+++ b/option_test.go
@@ -68,6 +68,11 @@ func TestOptionCases(t *testing.T) {
                "(imported.oss.package).action",
                "literal-double-quotes",
                "",
+       }, {
+               `option (imported.oss.package).action = "key:\"literal-double-quotes-escapted\"";`,
+               "(imported.oss.package).action",
+               `key:\"literal-double-quotes-escaped\"`,
+               "",
        }, {
                `option (imported.oss.package).action = 'literalsinglequotes';`,
                "(imported.oss.package).action",
```

This is the result:

```
$ go test .
--- FAIL: TestOptionCases (0.00s)
    option_test.go:101: [7] got [key:\"literal-double-quotes-escapted\] want [key:\"literal-double-quotes-escaped\"]
```

--- a/option_test.go
+++ b/option_test.go
@@ -69,6 +69,11 @@ func TestOptionCases(t *testing.T) {
 		"literal-double-quotes",
 		"",
 	}, {
+		`option (imported.oss.package).action = "key:\"literal-double-quotes-escaped\"";`,
+		"(imported.oss.package).action",
+		`key:\"literal-double-quotes-escaped\"`,
+		"",
+	}, {
 		`option (imported.oss.package).action = 'literalsinglequotes';`,
 		"(imported.oss.package).action",
 		"literalsinglequotes",
@@ -581,7 +586,7 @@ func TestOptionWithRepeatedMessageValues(t *testing.T) {
 
 func TestOptionWithRepeatedMessageValuesWithArray(t *testing.T) {
 	src := `message Foo {
-		int64 a = 1 [ (bar.repeated_field_dep_option) = 
+		int64 a = 1 [ (bar.repeated_field_dep_option) =
 			{ hello: 1, repeated_dep: [
 				{ hello: 1, repeated_bar: [1, 2] },
 				{ hello: 3, repeated_bar: [3, 4] } ] } ];

--- a/token.go
+++ b/token.go
@@ -120,7 +120,24 @@ func isComment(lit string) bool {
 }
 
 func unQuote(lit string) string {
-	return strings.Trim(lit, "\"'")
+	lit = strings.TrimLeft(lit, "\"'")
+	array := []rune(lit)
+	// https://github.com/emicklei/proto/issues/103
+	// cannot use strconv.Unquote as this unescapes quotes
+	for len(array) > 0 {
+		last := array[len(array)-1]
+		if last != '\'' && last != '"' {
+			break
+		}
+		if len(array) > 1 {
+			secondLast := array[len(array)-2]
+			if secondLast == '\\' {
+				break
+			}
+		}
+		array = array[:len(array)-1]
+	}
+	return string(array)
 }
 
 func asToken(literal string) token {

